test/unittest/test_lib_utils.py

```diff
@@ -0,0 +1,36 @@
+import unittest
+
+import socket
+
+from hypothesis import given, settings
+from hypothesis.strategies import integers
+
+import lib.utils as utils
+
+
+class TestUtils(unittest.TestCase):
+    def test_extract_schema_from_snapshot(self):
+        snapshot_path = '00000000000000000003.snap'
+        v = utils.extract_schema_from_snapshot(snapshot_path)
+        # assertListEqual() is available since 3.1
+        is_equal = v == [u'version', 2, 3, 1]
```

Comment:
(Again, not about the test per se. Feel free to postpone.)

There are two points that I dislike about the API of the function:

1. It returns a unicode string unlike all other test-run code (which sticks to `str`).
2. It would be better to return just version tuple, like `(2, 3, 1)` -- quite usual for representing a version in Python. And it works nice with less-great comparisons.

The first could be resolved by using pyyaml instead of json:

```python
# (python 2)
>>> import json
>>> import yaml
>>> json.loads('["version", 1, 2, 3]')
[u'version', 1, 2, 3]
>>> yaml.safe_load('["version", 1, 2, 3]')
['version', 1, 2, 3]
```

However if we'll implement the second point, it is not necessary. It is quite simple:

```diff
diff --git a/lib/tarantool_server.py b/lib/tarantool_server.py
index 45c5148..a0ad6da 100644
--- a/lib/tarantool_server.py
+++ b/lib/tarantool_server.py
@@ -912,8 +912,8 @@ class TarantoolServer(Server):
         # Verify that the schema actually was not upgraded.
         if self.disable_schema_upgrade:
             expected_version = extract_schema_from_snapshot(self.snapshot_path)
-            actual_version = yaml.safe_load(self.admin.execute(
-                'box.space._schema:get{"version"}'))[0]
+            actual_version = tuple(yaml.safe_load(self.admin.execute(
+                'box.space._schema:get{"version"}'))[0][1:])
             if expected_version != actual_version:
                 color_stdout('Schema version check fails: expected '
                              '{}, got {}\n'.format(expected_version,
diff --git a/lib/utils.py b/lib/utils.py
index a92db07..f88ebb5 100644
--- a/lib/utils.py
+++ b/lib/utils.py
@@ -303,7 +303,7 @@ def extract_schema_from_snapshot(snapshot_path):
        "BODY": {"space_id": 272, "tuple": ["version", 2, 3, 1]}
      }
 
-    :returns: [u'version', 2, 3, 1]
+    :returns: (2, 3, 1)
     """
     BOX_SCHEMA_ID = 272
     for row in xlog_rows(snapshot_path):
@@ -311,7 +311,7 @@ def extract_schema_from_snapshot(snapshot_path):
            row['BODY']['space_id'] == BOX_SCHEMA_ID:
             res = row['BODY']['tuple']
             if res[0] == 'version':
-                return res
+                return tuple(res[1:])
     return None
```

Reply:
Applied patch, thanks!